File: fix_paths.js
const fs = require('fs');
const path = require('path');

function replaceInFile(filePath, searchStr, replaceStr) {
    if (fs.existsSync(filePath)) {
        let content = fs.readFileSync(filePath, 'utf8');
        content = content.replace(searchStr, replaceStr);
        fs.writeFileSync(filePath, content, 'utf8');
    }
}

function replaceAllInFile(filePath, searchPattern, replaceStr) {
    if (fs.existsSync(filePath)) {
        let content = fs.readFileSync(filePath, 'utf8');
        content = content.replace(new RegExp(searchPattern, 'g'), replaceStr);
        fs.writeFileSync(filePath, content, 'utf8');
    }
}

// 1. Root Links
replaceInFile('index.html', 'src="app.js"', 'src="js/app.js"');
replaceInFile('auth.html', 'src="app.js"', 'src="js/app.js"');
replaceInFile('auth.html', 'src="auth.js"', 'src="js/auth.js"');

// 2. Auth JS Redirect
replaceInFile('js/auth.js', '-dashboard.html', 'dashboards/{user.role}-dashboard.html');

// 3. App JS Redirect (Logout)
replaceInFile('js/app.js', 'window.location.href = \'auth.html\';', 'window.location.href = window.location.pathname.includes(\'dashboards/\') ? \'../auth.html\' : \'auth.html\';');

// 4. Dashboard JS Redirects
['donor', 'receiver', 'volunteer'].forEach(role => {
    const jsPath = 'js/' + role + '-dashboard.js';
    replaceInFile(jsPath, 'window.location.href = \'auth.html\';', 'window.location.href = \'../auth.html\';');
    
    // Replace URL paths in HTML
    const htmlPath = 'dashboards/' + role + '-dashboard.html';
    replaceInFile(htmlPath, 'href="styles.css"', 'href="../styles.css"');
    replaceInFile(htmlPath, 'href="index.html"', 'href="../index.html"');
    replaceInFile(htmlPath, 'src="app.js"', 'src="../js/app.js"');
    replaceInFile(htmlPath, 'src="' + role + '-dashboard.js"', 'src="../js/' + role + '-dashboard.js"');
});

console.log('Update paths completed for ' + process.cwd());

File: fix_paths.py
import os
import re

def repl_file(path, old, new):
    if os.path.exists(path):
        with open(path, 'r', encoding='utf-8') as f:
            content = f.read()
            content = content.replace(old, new)
        with open(path, 'w', encoding='utf-8') as f:
            f.write(content)

# Root files
repl_file('index.html', 'src="app.js"', 'src="js/app.js"')
repl_file('auth.html', 'src="app.js"', 'src="js/app.js"')
repl_file('auth.html', 'src="auth.js"', 'src="js/auth.js"')

# JS files
repl_file('js/auth.js', '-dashboard.html', 'dashboards/-dashboard.html')
repl_file('js/app.js', 'window.location.href = \'auth.html\';', 'window.location.href = window.location.pathname.endswith(\\\'auth.html\\\') || window.location.pathname.endswith(\\\'index.html\\\') ? \\\'auth.html\\\' : \\\'../auth.html\\\';')
repl_file('js/donor-dashboard.js', 'window.location.href = \'auth.html\';', 'window.location.href = \'../auth.html\';')
repl_file('js/receiver-dashboard.js', 'window.location.href = \'auth.html\';', 'window.location.href = \'../auth.html\';')
repl_file('js/volunteer-dashboard.js', 'window.location.href = \'auth.html\';', 'window.location.href = \'../auth.html\';')

# HTML Dashboards
for p in ['dashboards/donor-dashboard.html', 'dashboards/receiver-dashboard.html', 'dashboards/volunteer-dashboard.html']:
    repl_file(p, 'href="styles.css"', 'href="../styles.css"')
    repl_file(p, 'href="index.html"', 'href="../index.html"')
    repl_file(p, 'src="app.js"', 'src="../js/app.js"')
    dl = os.path.basename(p).replace(".html", ".js")
    repl_file(p, f'src="{dl}"', f'src="../js/{dl}"')

print('Done reorganizing and updating paths!')
